Fix Prometheus additional configuration documentation (#3505)
strimzi/strimzi-kafka-operator: documentation/modules/metrics/proc_metrics-deploying-prometheus.adoc

. Edit the `PodMonitor` resource in `strimzi-pod-monitor.yaml` to define Prometheus jobs that will scrape the metrics data from pods.
`PodMonitor` is used to scrape data directly from pods and is used for Apache Kafka, ZooKeeper, Operators, and Kafka Bridge. Update the `namespaceSelector.matchNames` property with the namespace where the pods to scrape the metrics from are running.

. The provided Grafana dashboards show metrics about CPU, memory and volume disks usage.
These metrics come from cadvisor and kubelet directly on the nodes.
In order to have Prometheus scraping for these metrics, an additional configuration is needed.

> This is because the Prometheus operator doesn't have a corresponding of `PodMonitor` for scraping metrics coming from nodes directly.

Reviewer: ```suggestion
This is because the Prometheus operator does not have a corresponding of `PodMonitor` for scraping metrics coming from nodes directly.
```

Author: Thanks Standa!

> . The provided Grafana dashboards show metrics about CPU, memory and volume disks usage.
> These metrics come from cadvisor and kubelet directly on the nodes.
> In order to have Prometheus scraping for these metrics, an additional configuration is needed.
> This is because the Prometheus operator does not have a corresponding of `PodMonitor` for scraping metrics coming from nodes directly.

Reviewer 2: ```suggestion
+
The Grafana dashboards provided show metrics for CPU, memory and disk volume usage,
which come directly from the Kubernetes cAdvisor agent and kubelet on the nodes.
The configuration is needed because the Prometheus operator does not have a corresponding `PodMonitor` for scraping metrics directly from nodes.
```

Reviewer 2: I'm assuming this information relates to the step above: . Edit the `PodMonitor` resource

Author: No it's a step on its own.

> .. Create a `Secret` resource:

Reviewer 2: Should this sub-step and the next sub-step be steps now? We removed the _To use another role:_ step that grouped them

Author: I am not removing step 3. The sentence "To use another role" doesn't make any sense without a context. I guess it's referring to the role "node" used in the Prometheus configuration for scraping the metrics from cadvisor and kubelet but the user don't need to dig into it. My explanation should be enough.
